### PR TITLE
Issue 14059 - Formatted write with wrong formatting string

### PR DESCRIPTION
**Pull request by sinkuu**
https://issues.dlang.org/show_bug.cgi?id=14059

I adjusted enforce condition, because it reads two(or three if '-' exists) characters in one loop.

### CODE REVIEW

**Comment by burner** [comment]
could you and a unittest to test the snipped from bugzilla

**Comment by sinkuu** [comment]
Changed (testing both `FormatSpec` and `writefln` seems redundant) to use `writefln`.

**Comment by burner** [comment]
maybe I was not clear. Write the array(red,blue) into an appender and check if the data of the appender is what you expect. writeln in unittests isn't nice. Its just noice on stdout.

**Comment by sinkuu** [comment]
Well, the test case is a invalid format spec that should fail (please see the issue). At first I tested FormatSpec directly because I think that only parsing format specs is needed.

**Comment by burner** [comment]
I see my mistake

**Comment by burner** [comment]
LGTM

File: std/format.d [hunk 1]
```diff
@@ -867,7 +867,7 @@ struct FormatSpec(Char)
                 // Get the matching balanced paren
                 for (uint innerParens;;)
                 {
-                    enforce(j < trailing.length,
+                    enforce(j + 1 < trailing.length,
```
Comment (JakobOvrum) [inline std/format.d]: Maybe this should throw `FormatException`, not `Exception`?

Description of `FormatException`:
> Signals a mismatch between a format and its corresponding argument.

Not quite what is going on here... maybe a new exception type for misformed format strings?

Comment (sinkuu) [inline std/format.d]: `enforceFmt` (= `enforce!FormatException`) is already used for indicating format string errors. I'll make use of this.

**Comment by JakobOvrum** [comment]
LGTM

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Comment by sinkuu** [comment]
Thank you!

**Pull request merged**